packages/webapp/components/history/search.tsx

```diff
@@ -1,5 +1,60 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useMemo } from 'react';
+import { useInfiniteQuery } from 'react-query';
+import {
+  getSearchHistory,
+  getSearchIdUrl,
+  SearchHistoryData,
+} from '@dailydotdev/shared/src/graphql/search';
+import {
+  generateQueryKey,
+  RequestKey,
+} from '@dailydotdev/shared/src/lib/query';
+import user from '@dailydotdev/shared/__tests__/fixture/loggedUser';
+import { SearchBarSuggestion } from '@dailydotdev/shared/src/components';
+import useFeedInfiniteScroll, {
+  InfiniteScrollScreenOffset,
+} from '@dailydotdev/shared/src/hooks/feed/useFeedInfiniteScroll';
+import { SearchSkeleton } from './SearchSkeleton';
+import { SearchEmpty } from './SearchEmpty';
+import { SearchHistoryContainer } from './common';
 
 export function SearchHistory(): ReactElement {
-  return <div className="flex flex-col gap-3">SearchHistoryList</div>;
+  const { data, isLoading, fetchNextPage, isFetchingNextPage, hasNextPage } =
+    useInfiniteQuery<SearchHistoryData>(
+      generateQueryKey(RequestKey.SearchHistory, user),
+      ({ pageParam }) => getSearchHistory({ after: pageParam, first: 30 }),
+      {
+        getNextPageParam: (lastPage) =>
+          lastPage?.history?.pageInfo.hasNextPage &&
+          lastPage?.history?.pageInfo.endCursor,
+      },
+    );
+
+  const canFetchMore =
+    !isLoading && !isFetchingNextPage && hasNextPage && data.pages.length > 0;
```

Comment:
Nothing to action on, but maybe web discussion, can we not add this to `useInifniteQuery` or wrap it, we always use the same

Reply:
I agree, we should probably create a hook to abstract these requirements. Kinda messy to copy paste it everywhere.

Reply:
Cool if it's easy feel free to do so, if not I can log it on the backlog

Reply:
Seems like there is more work involved relating to TypeScript. I think we can create a ticket and include the retro updates for existing implementations. Wdyt?